psst-gui/src/webapi/client.rs

```diff
@@ -707,7 +1143,7 @@ impl WebApi {
 impl WebApi {
     // https://developer.spotify.com/documentation/web-api/reference/get-audio-analysis/
     pub fn _get_audio_analysis(&self, track_id: &str) -> Result<AudioAnalysis, Error> {
-        let request = self.get(format!("v1/audio-analysis/{}", track_id))?;
+        let request = self.get(format!("v1/audio-analysis/{}", track_id), None)?;
```

Comment:
Not sure how this would work in Rust but can optional args be specified that take a default if they aren't included in the call? This is looking a little messy to change every single request.

Reply:
I was trying to find this out too. Just had a deeper look, function overloading needs an extra crate: https://docs.rs/overloadf/latest/overloadf/#:~:text=With%20a%20single%20attribute%20on,use%20at%20your%20own%20risk.&text=There%20are%20some%20features%20that,unsafe%20function%20overloading

Reply:
However, this library is based on some unstable features. Do you think this is ok? 

It does not seem to be very supported in rust: https://internals.rust-lang.org/t/justification-for-rust-not-supporting-function-overloading-directly/7012

Reply:
We shouldn't add any new crates unless they are small and absolutely necessary I think. If this is non-standard in Rust then this is fine.